src/main/java/seedu/address/model/recommendation/UniqueRecommendationList.java

```diff
@@ -0,0 +1,144 @@
+package seedu.address.model.recommendation;
+
+import static java.util.Objects.requireNonNull;
+import static seedu.address.commons.util.CollectionUtil.requireAllNonNull;
+
+import java.util.Iterator;
+import java.util.List;
+
+import javafx.collections.FXCollections;
+import javafx.collections.ObservableList;
+import seedu.address.model.recommendation.exceptions.DuplicateRecommendationException;
+import seedu.address.model.recommendation.exceptions.RecommendationNotFoundException;
+
+/**
+ * A list of recommendations that enforces uniqueness between its elements and does not allow nulls.
+ * A recommendation is considered unique by comparing using
+ * {@code Recommendation#isSameRecommendation(Recommendation)}. As such, adding and updating of
+ * recommendations uses Recommendation#isSameRecommendation(Recommendation)
+ * for equality so as to ensure that the recommendation being added or updated is
+ * unique in terms of identity in the UniqueRecommendationList.
+ * However, the removal of a recommendation uses Recommendation#equals(Object) so
+ * as to ensure that the recommendation with exactly the same fields will be removed.
+ *
+ * Supports a minimal set of list operations.
+ *
+ * @see Recommendation#isSameRecommendation(Recommendation)
+ */
```

Comment:
Header is somehow lengthy, and it explains the WHAT and WHY, not HOW

Reply:
Do also not we should exclude explanation for how the code works. It should have been apparent in the code.

Reply:
Hmm... This explanation is actually what's inside `UniquePersonList` so you likely have a problem with the original AB3 explanation. **This is purely copy-paste**.